Support customizing read timeout and sleeps.

iwanders opened the pull request:
This should implement #5.

It adds:
- `--read-timeout`, a timeout on the pyUSB `.read()` call in milliseconds. Can be set through the environment flag `GPSPOD_READ_TIMEOUT`. (Add `export GPSPOD_READ_TIMEOUT=1000` to `.bashrc`).
- `--read-sleep-minsize`, if the packet exceeds this size the code will sleep for `read-sleep-duration`. In bytes, can be set through the environment flag `GPSPOD_READ_SLEEP_MINSIZE`.
- `--read-sleep-duration` The duration to sleep if the packet exceeded the minsize. Can be set through the environment flag `GPSPOD_READ_SLEEP_DURATION`), in milliseconds.

For the HIDAPI the following exports are required to preserve old behaviour.
```
export GPSPOD_READ_SLEEP_MINSIZE=0
export GPSPOD_READ_SLEEP_DURATION=0.1
```

@inancme you asked for this in #5, can you give it a go and experiment with which values work best on your Raspberry Pi? I'll add the values you recommend to the readme, detailing the issue on the raspberry pi and recommended values to overcome timeouts.

iwanders committed:
Support customizing read timeout and sleeps.

inancme reviewed (changes requested):
Perhaps Line 479 should contain:
```Python
                default=read_sleep_duration_val, type=float)
```
The sleep duration was not being propagated to the Communicator.

inancme commented:
> This should implement #5.
> 
> It adds:
> 
> * `--read-timeout`, a timeout on the pyUSB `.read()` call in milliseconds. Can be set through the environment flag `GPSPOD_READ_TIMEOUT`. (Add `export GPSPOD_READ_TIMEOUT=1000` to `.bashrc`).
> * `--read-sleep-minsize`, if the packet exceeds this size the code will sleep for `read-sleep-duration`. In bytes, can be set through the environment flag `GPSPOD_READ_SLEEP_MINSIZE`.
> * `--read-sleep-duration` The duration to sleep if the packet exceeded the minsize. Can be set through the environment flag `GPSPOD_READ_SLEEP_DURATION`), in milliseconds.
> 
> For the HIDAPI the following exports are required to preserve old behaviour.
> 
> ```
> export GPSPOD_READ_SLEEP_MINSIZE=0
> export GPSPOD_READ_SLEEP_DURATION=0.1
> ```
> 
> @inancme you asked for this in #5, can you give it a go and experiment with which values work best on your Raspberry Pi? I'll add the values you recommend to the readme, detailing the issue on the raspberry pi and recommended values to overcome timeouts.

I could test it with Raspberry Pi 2 and the values that worked are:
export GPSPOD_READ_SLEEP_MINSIZE=10000
export GPSPOD_READ_SLEEP_DURATION=1000

iwanders committed:
Actually populate the sleep duration with the value.

iwanders committed:
Update readme.

iwanders commented:
> Perhaps Line 479 should contain:

Yes it should have! Glad you spotted that, thanks for the code review, fixed in e815deb54bb0ab3d8c11f3b7a7e22ae6796267c7.

> I could test it with Raspberry Pi 2 and the values that worked are:

Thanks, I've incorporated this in the readme in 6bdcd60a02a5fcb011b147651ed95300b59f31bb, pretty render [here](https://github.com/iwanders/gps_track_pod/tree/configurable-usb-timeouts#usb-timeouts--raspberry-pi).

Let me know if you think this is in a good state to go in.

iwanders commented:
I'm merging this, confirmed today again that it works, values taken from the comment. Thanks for contributing @inancme .
